File: src/main/java/com/nurkiewicz/tsclass/parser/ast/Type.java
package com.nurkiewicz.tsclass.parser.ast;

import lombok.Value;

@Value
public class Type {

    private final String name;

    public org.objectweb.asm.Type toJavaType() {
        switch (name) {
            case "number":
                return org.objectweb.asm.Type.DOUBLE_TYPE;
            case "string":
                return org.objectweb.asm.Type.getType(String.class);
            default:
                throw new IllegalArgumentException("Uknown type: " + name);
        }
    }

}

File: src/main/kotlin/com/nurkiewicz/tsclass/parser/ast/Type.kt
package com.nurkiewicz.tsclass.parser.ast

data class Type (val name: String, val stackSize: Int) {

    companion object {
        val number = Type("number", 2)
    }

    fun toJavaType(): org.objectweb.asm.Type {
        when (name) {
            "number" -> return org.objectweb.asm.Type.DOUBLE_TYPE
            "string" -> return org.objectweb.asm.Type.getType(String::class.java)
            else -> throw IllegalArgumentException("Uknown type: " + name)
        }
    }

    override fun toString() = name
}
